.github/workflows/moodle-plugin-ci.yml

```diff
@@ -0,0 +1,119 @@
+          MYSQL_ALLOW_EMPTY_PASSWORD: "true"
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval 10s --health-timeout 5s --health-retries 3
+
+    # PHP Compatibility Matrix Moodle           End of life
+    # Moodle 3.8: php7.1, php7.2, php7.3        10 May 2021
+    # Moodle 3.9: php7.2, php7.3, php7.4        8 May 2023
+    # Moodle 3.10: php7.2, php7.3, php7.4       9 May 2022
+    # earlier versions of Moodle and php are not included, as support was dropped
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Overall strategy is to run as few jobs as possible, but:
+        # - each php version at least once
+        # - each moodle version at least once
+        # - each database at least once
+        include:
+          # php edge cases with pgsql
+          - {php: '7.1', moodle-branch: MOODLE_38_STABLE, database: pgsql}
+          - {php: '7.2', moodle-branch: MOODLE_39_STABLE, database: mariadb}
+          - {php: '7.3', moodle-branch: MOODLE_39_STABLE, database: pgsql}
+          - {php: '7.4', moodle-branch: MOODLE_310_STABLE, database: mariadb}
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+        with:
+          path: plugin
+
+      - name: Setup PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          coverage: none
+
+      - name: Initialise moodle-plugin-ci
+        run: |
+          composer create-project -n --no-dev --prefer-dist moodlehq/moodle-plugin-ci ci ^3
+          echo $(cd ci/bin; pwd) >> $GITHUB_PATH
+          echo $(cd ci/vendor/bin; pwd) >> $GITHUB_PATH
+          sudo locale-gen en_AU.UTF-8
+          echo "NVM_DIR=$HOME/.nvm" >> $GITHUB_ENV
+
+      - name: Install moodle-plugin-ci
+        run: |
+          moodle-plugin-ci install --plugin ./plugin --db-host=127.0.0.1
+        env:
+          DB: ${{ matrix.database }}
+          MOODLE_BRANCH: ${{ matrix.moodle-branch }}
+
+      - name: PHP Lint
+        if: ${{ always() }}
+        run: moodle-plugin-ci phplint
+
+      - name: PHP Copy/Paste Detector
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpcpd
+
+      - name: PHP Mess Detector
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpmd
+
+      - name: Moodle Code Checker
+        if: ${{ always() }}
+        run: moodle-plugin-ci codechecker --max-warnings 0
+        continue-on-error: true
+
+      - name: Moodle PHPDoc Checker
+        if: ${{ always() }}
+        run: moodle-plugin-ci phpdoc
+        continue-on-error: true
+
+      - name: Validating
+        if: ${{ always() }}
+        run: moodle-plugin-ci validate
+
+      - name: Check upgrade savepoints
+        if: ${{ always() }}
+        run: moodle-plugin-ci savepoints
+
+      - name: Mustache Lint
+        if: ${{ always() }}
+        run: moodle-plugin-ci mustache
+        continue-on-error: true
```

Comment:
With Github Actions mustache checker seems to run correctly. I'm not familiar with mustache at all, is maybe someone able to have a quick look at those files which error with `WARNING: Example context missing`:

https://github.com/frankkoch/moodle-mod_studentquiz/blob/master/templates/digest_email_notification.mustache
https://github.com/frankkoch/moodle-mod_studentquiz/blob/master/templates/single_email_notification.mustache

Reply:
In the comment at the top of the template, you should include, as JSON code, the type of date that might be used to render the template. E.g. https://github.com/moodleou/moodle-filter_embedquestion/blob/main/templates/embed_iframe.mustache#L30

You can see the effect of that at Admin -> Development -> Template library in a Moodle install.

Reply:
Thank you Tim! I'll have a look at all your suggestions, but I'm now out of time for today.